backends-common/cassandra/src/main/java/org/apache/james/backends/cassandra/utils/CassandraUtils.java

```diff
@@ -21,13 +21,44 @@
 
 import com.datastax.driver.core.ResultSet;
 import com.datastax.driver.core.Row;
+import com.datastax.driver.core.Session;
+import com.datastax.driver.core.Statement;
+
+import java.util.Optional;
+import java.util.concurrent.CompletableFuture;
 import java.util.stream.Stream;
 import java.util.stream.StreamSupport;
 
+import net.javacrumbs.futureconverter.java8guava.FutureConverter;
+
 public class CassandraUtils {
 
     public static Stream<Row> convertToStream(ResultSet resultSet) {
         return StreamSupport.stream(resultSet.spliterator(), true);
     }
 
+    public static CompletableFuture<ResultSet> executeAsync(Session session, Statement statement) {
```

Comment:
All added methods are about async execution, maybe it could a grouped into a AsyncStatementExecutor object.
We could pass the session into the constructor to avoid repeating ourselves and shorten method names because most of the semantic would be in the class name